packages/block-editor/src/components/inserter/quick-inserter.js

```diff
@@ -46,26 +47,39 @@ export default function QuickInserter( {
 		onInsertBlocks,
 		destinationRootClientId
 	);
-	const showPatterns = patterns.length && !! filterValue;
-	const showSearch =
-		( showPatterns && patterns.length > SEARCH_THRESHOLD ) ||
-		blockTypes.length > SEARCH_THRESHOLD;
 
-	const { setInserterIsOpened, insertionIndex } = useSelect(
+	const {
+		setInserterIsOpened,
+		insertionIndex,
+		prioritizePatterns,
+	} = useSelect(
 		( select ) => {
 			const { getSettings, getBlockIndex, getBlockCount } = select(
 				blockEditorStore
 			);
+			const settings = getSettings();
 			const index = getBlockIndex( clientId );
+			const blockCount = getBlockCount();
+
 			return {
-				setInserterIsOpened: getSettings()
-					.__experimentalSetIsInserterOpened,
-				insertionIndex: index === -1 ? getBlockCount() : index,
+				setInserterIsOpened: settings.__experimentalSetIsInserterOpened,
+				prioritizePatterns:
+					settings.__experimentalPrioritizePatternsOnQuickInserterRoot &&
+					! rootClientId &&
+					index > 0 &&
```

Comment:
What about these positions? If I create a new empty template I should see patterns in the inserter, no? Same goes for the last index if it's top level.

Reply:
Hi @ntsekouras,
The conditions are following what @mtias specified on the issue:
> The top-level means it's a direct descendant of the document, between any root template parts or existing blocks

I added a condition to also show the behavior when the template is empty.

Reply:
Yes, we'll need to do some extra work on empty templates regardless.